schemas/processing.json

```diff
@@ -0,0 +1,75 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "type": "object",
+    "properties": {
+        "describedBy": {
+            "type": "string",
+            "const": "https://github.com/AllenNeuralDynamics/data_schema/blob/main/schemas/processing.json",
+            "description": "The URL reference to the schema.",
+            "title": "Described by"
+        },
+        "version": {
+            "type": "string",
+            "const": "0.0.1",
+            "description": "schema version",
+            "title": "Version"
+        },
+        "data_processing": {
+            "type": "array",
+            "items": { "$ref": "#/$defs/processes" },
+            "uniqueItems": true,
+            "title": "Data processing"
+        }
+    },
+    "required": [
+        "describedBy",
+        "version",
+        "data_processing"
+    ],
+    "$defs": {
+        "processes":{
+            "type": "object",
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "title": "Name"
+                },
+                "version": {
+                    "type": "string",
+                    "title": "Version",
+                    "description": "Version of the algorithm used"
+                },
+                "date_time": {
+                    "type": "string",
+                    "format": "date-time"
+                },
+                "data_inputs": {
+                    "type": "string",
+                    "title": "Data inputs",
+                    "description": "Path to data inputs"
+                },
+                "data_outputs": {
+                    "type": "string",
+                    "title": "Data outputs",
+                    "description": "Path to data outputs"
+                },
+                "respository": {
+                    "type": "string",
+                    "title": "Respository",
+                    "description": "Path to code respository"
+                },
+                "notes": {
+                    "type": "string"
+                }
+            },
+            "required": [
```

Comment:
I think I have the same reaction to this as the time stamp. Unless the version/repository applies to *all* the processes, this will be confusing. 
 But is there something we can add to the subschema to indicate when things are part of a single capsule?

Reply:
It's not that the version applies to everything - it's that the definition of "this collection of subprocesses" is itself versioned. SmartSpim Stitcher Capsule 1.0 is a combination of pystripe 1.2, terastitcher 2.3, and omeconverter 3.4.  But maybe we swap out terastitcher 2.3 with terastitcher 3.4 at some point.  That becomes SmartSPIM Stitcher Capsule 1.1.  

It's a bit like how the rig has a version but is made up of parts with their own serial numbers.

Reply:
But this only works if all of the processing in the json is part of that capsule.